core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/federation/JoinExecutorBase.java

```diff
@@ -24,20 +25,29 @@
  */
 public abstract class JoinExecutorBase<T> extends LookAheadIteration<T, QueryEvaluationException> {
 
+	/**
+	 * @deprecated No replacement, don't use static shared int variables.
+	 */
 	protected static int NEXT_JOIN_ID = 1;
```

Comment:
If this variable is actually used, it should be a static final AtomicInteger rather than a static mutable int. There were no references to it, but still deprecating it rather than changing or removing it as protected variables are presumed to be part of the public API until we clarify the procedure for these cases.